### PR TITLE
New Code Example: Wavelet CNN for texture classification

### PR DESCRIPTION
**Pull request by lucifertrj**
This is a code example, I have implemented **Wavelet CNN for Texture Classification**. Wavelet CNN is based on Wavelet Transform in Image Processing. This paper demonstrates how to build a Wavelet CNN model. [https://arxiv.org/pdf/1805.08620.pdf](https://arxiv.org/pdf/1805.08620.pdf)

@fchollet Could you please look into it?

Thank you

### CODE REVIEW

**Comment by lucifertrj** [comment]
@fchollet, any changes to be made?

**Comment by lucifertrj** [comment]
@fchollet, May I know why is still merging blocked?

**Comment by fchollet** [comment]
Sorry for the delay in reviewing! Thank you for your patience.

Overall this PR suffers from a number of issues and isn't up to our usual standard. Some of the issues include:

- There is no text whatsoever to explain to the reader what the example does, what the different components are, etc.
- Reliance on deprecated API symbols such as `ImageDataGenerator`
- Keras code examples are normally around the lines of "how to use Keras APIs to solve use case X", but here the only up to date Keras API is a simple convnet (which is already covered by other code examples), the remainder is ad hoc low-level code (the wavelet transforms).
- All of the ad hoc code was taken from an [external source](https://github.com/menon92/WaveletCNN/). The source is linked in the PR code, which is good, but it's not a good practice to just take people's OSS code and repackage it with little value add.
- Reliance on external libraries for this that one can do with Keras APIs, e.g. ShuffleSplit from sklearn
- Code style is not formatted (next time please read [the contributor guide](https://github.com/keras-team/keras-io/blob/master/contributor_guide.md) first)

We will not merge, sorry.
